internal/cmd/server.go

```diff
@@ -108,21 +110,24 @@ func getConfigByPath(cfgPath string) (*config.Server, error) {
 
 // setupRunners starts all the runners required for the Envoy Gateway to
 // fulfill its tasks.
-func setupRunners(cfg *config.Server) error {
+func setupRunners(cfg *config.Server) (err error) {
 	// TODO - Setup a Config Manager
 	// https://github.com/envoyproxy/gateway/issues/43
 	ctx := ctrl.SetupSignalHandler()
 
-	// Setup the Extension Manager
-	extMgr, err := extensionregistry.NewManager(cfg)
-	if err != nil {
-		return err
+	// Setup the Extension Manager for Kubernetes provider.
+	var extMgr types.Manager
+	if cfg.EnvoyGateway.Provider.Type == egv1a1.ProviderTypeKubernetes {
```

Comment:
curious why this is needed ? imo its not required

Reply:
the extension manager will init a k8s client, so it only should be running with k8s provider.